app/scripts/modules/core/src/account/account.service.ts

```diff
@@ -166,13 +166,16 @@ export class AccountService {
       });
   }
 
-  public listAccounts(provider: string = null): ng.IPromise<IAccount[]> {
-
+  public listAccounts(provider: string = null, providerVersion: string = null): ng.IPromise<IAccount[]> {
     let result: ng.IPromise<IAccount[]> = this.API.one('credentials').useCache().get();
     if (provider) {
       result = result.then((accounts: IAccount[]) => accounts.filter((account: IAccount) => account.type === provider));
     }
 
+    if (providerVersion) {
+      result = result.then((accounts: IAccount[]) => accounts.filter((account: IAccount) => account.providerVersion === providerVersion ));
```

Comment:
Extra space at the end: `account.providerVersion === providerVersion)`

Reply:
This isn't done above, but you should be able to omit the parameter typing in each arrow function here, since the types should be inferred from `let result: ng.IPromise<IAccount[]> = ... `

Reply:
SG, if I drop it here should I drop it above as well?

Reply:
Sure.